test/compress/harmony.js

```diff
@@ -583,3 +583,39 @@ class_extends_regex: {
     }
     expect_exact: "function f(){class rx1 extends(/rx/){}}"
 }
+
+issue_2028: {
+    options = {
+        side_effects: true,
```

Comment:
Thanks.

Although it matches node's behavior, which is what we want, I must admit I don't know why the `class` is retained below the assignment here when it is not for `return function() { return X; }; class X {}`